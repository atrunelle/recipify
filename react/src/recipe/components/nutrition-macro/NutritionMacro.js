import React, { Component } from 'react';
import NutrientsMacroChart from '../nutrients-macro-chart/NutrientsMacroChart';
import { Card, CardContent, Typography } from '@material-ui/core';

class NutritionMacro extends Component {
  render() {
    if (!this.props.totalNutrients.length) return '';

    return (
      <Card>
        <CardContent>
          <Typography
            gutterBottom 
            variant="headline"
            component="h2">
            Macro in recipe
          </Typography>          
          <Typography
            gutterBottom 
            paragraph
            component="p">
              Based on 4 calories per gram of carbs and protein, and 9 calories per gram of fat.
          </Typography>
          <Typography
            gutterBottom 
            paragraph
            component="p">
            Those numbers are approximation only, as the number of calories per macro change per ingredient and depend on quality and other factor.
          </Typography>
          <NutrientsMacroChart nutritionData={this.props.totalNutrients}></NutrientsMacroChart>
        </CardContent>
      </Card>
    );
  }
}

export default NutritionMacro;