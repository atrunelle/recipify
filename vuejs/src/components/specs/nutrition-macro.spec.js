import { mount } from '@vue/test-utils';
import NutritionMacro from '../nutrition-macro';

describe('Component: nutrients chart', () => {
  it('should match snapshot', () => {
    const wrapper = mount(NutritionMacro, {
      propsData: {
        totalNutrients: [{
          totalQuantity: {
            label: 'Protein',
            percentage: 50,
          },
        }, {
          totalQuantity: {
            label: 'Fat',
            percentage: 10,
          },
        }],
      },
    });
  });
});
