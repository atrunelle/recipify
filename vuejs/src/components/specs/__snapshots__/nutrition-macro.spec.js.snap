// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: nutrients chart should match snapshot 1`] = `
<div class="flex xs12">
  <div class="v-card v-sheet theme--light">
    <div class="v-card__title">
      <h2 class="headline">
        Macro in recipe
      </h2>
    </div>
    <div class="v-card__text">
      <p class="body-1">
        Based on 4 calories per gram of carbs and protein, and 9 calories per gram of fat.
      </p>
      <p class="body-1">
        Those numbers are approximation only, as the number of calories per macro change per ingredient and depend on quality and other factor.
      </p>
      <div class="layout row">
        <div class="bar-graph  bar-graph--protein" style="width: 50%;">
          <p class="bar-graph__text">
            Protein 50%
          </p>
        </div>
        <div class="bar-graph  bar-graph--fat" style="width: 10%;">
          <p class="bar-graph__text">
            Fat 10%
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
